src/elements/video/generator.js

```diff
@@ -0,0 +1,226 @@
+      };
+    }];
+  }
+
+  function Generator(definition) {
+    if (!definition.source) {
+      throw new Error('Source is missing from generator definition');
+    }
+    if (!definition.embedUrl) {
+      throw new Error('Embed url is missing from generator definition');
+    }
+    angular.extend(this, definition);
+  }
+
+  Generator.prototype.generateUrl = function(params) {
+    var finalParams = {}, url = '';
+
+    if (this.defaultParams) {
+      angular.extend(finalParams, this.defaultParams);
+    }
+    angular.extend(finalParams, params);
+    angular.forEach(finalParams, function(value, key) {
+       url += '&amp;' + key + '=' + value;
+    });
+
+    return url;
+  };
+
+  Generator.prototype.generate = function(id, params) {
+    var html = '' +
+      '<iframe src="'+ this.embedUrl + id + '?=' + this.generateUrl(params) + '"' +
+      ' width="100%" height="100%"' +
+      ' frameborder="0" webkitAllowFullScreen mozallowfullscreen allowFullScreen></iframe>';
+    return angular.element(html);
+  };
+
+  function smVideoYoutube() {
+    var defaultParams = {
+      modestbranding: 1,
+      cc_load_policy: 1,
+      autohide: 0
+    };
+
+    this.extendDefaultParams = function (value) {
+      angular.extend(defaultParams, value);
+    };
+
+    this.$get = ['$window', '$q', function($window, $q) {
+      var regExp = /^(?:https?:\/\/)?(?:www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
+
+      function getId(url) {
+        var matches = url.match(regExp);
+        return matches && matches[1];
+      }
+
+      function loadApi() {
+        if (!loadApi.promise) {
+          var defer = $q.defer();
+
+          if ($window.YT) {
+            defer.resolve();
+          } else {
+            var tag = $window.document.createElement('script');
+
+            tag.src = "https://www.youtube.com/iframe_api";
+            var firstScriptTag = $window.document.getElementsByTagName('script')[0];
+            firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+            $window.onYouTubeIframeAPIReady = function() {
+              defer.resolve();
+            };
+          }
+          loadApi.promise = defer.promise;
+        }
+        return loadApi.promise;
+      }
+
+      function getPlayer(element) {
+        var defer = $q.defer();
+        loadApi().then(function() {
+          new $window.YT.Player(element, {
+            events: {
+              onReady: function(event) {
+                defer.resolve(event.target);
+              }
+            }
+          });
+        });
+        return defer.promise;
+      }
+
+      return {
+        defaultParams: defaultParams,
+        source: 'youtube',
+        embedUrl: '//www.youtube.com/embed/',
+        getIdFromUrl: getId,
+        getPlayer: getPlayer
+      };
+    }];
+  }
+
+  function smVideoVimeo() {
+    var defaultParams = {};
+
+    this.extendDefaultParams = function (value) {
```

Comment:
@mxth I suggested only to rename method but not to change implementation as well so that it doesn't mean you will lose another params. What about `setCustomDefaultParams(params)` ?
```
defaultParams = { autoplay: 1, rel: 0 };
setCustomDefaultParams({ rel: 1 });
// defaultParams is { autoplay: 1, rel: 1 };
```